### PR TITLE
Eng 9323

### PR DESCRIPTION
**Pull request by manjujames**
Added one more parameter ProcTimeout to specify the new timeout value.

### CODE REVIEW

File: src/frontend/org/voltdb/HTTPClientInterface.java [hunk 1]
```diff
@@ -164,6 +164,27 @@ public void process(Request request, HttpServletResponse response) {
 
             String procName = request.getParameter("Procedure");
             String params = request.getParameter("Parameters");
+            String timeoutStr = request.getParameter("ProcTimeout");
```
Comment (akhanzode) [inline src/frontend/org/voltdb/HTTPClientInterface.java]: Check with Andrew what name should be.

File: src/frontend/org/voltdb/HTTPClientInterface.java [hunk 2]
```diff
@@ -164,6 +164,27 @@ public void process(Request request, HttpServletResponse response) {
...
+            int procCallTimeout = -1;
+            if (timeoutStr != null) {
+                try {
+                    procCallTimeout = Integer.parseInt(timeoutStr);
+                    if (procCallTimeout <= 0) {
+                        throw new NumberFormatException();
```
Comment (akhanzode) [inline src/frontend/org/voltdb/HTTPClientInterface.java]: Do you want to abort or not use bad value ? or use default?

File: src/frontend/org/voltdb/HTTPClientInterface.java [hunk 3]
```diff
@@ -180,10 +201,10 @@ public void process(Request request, HttpServletResponse response) {
                     msg = String.format("%s( %s )", jsonp, msg);
                 }
                 response.setStatus(HttpServletResponse.SC_OK);
-                request.setHandled(true);
                 try {
                     response.getWriter().print(msg);
                 } catch (IOException e1) {} // Ignore this as browser must have closed.
+                request.setHandled(true);
```
Comment (akhanzode) [inline src/frontend/org/voltdb/HTTPClientInterface.java]: move it in finally to be safe?

File: src/frontend/org/voltdb/HTTPClientInterface.java (hunk 2 above)
Comment (manjujames) [inline src/frontend/org/voltdb/HTTPClientInterface.java]: They passed in a bad argument value. I think its better to abort than silently use a default value.

**Comment by manjujames** [comment]
Squash merged.

**Pull request closed without merging**